### ISSUE TITLE
The dollar ($) prefixed field '$inc' in '$inc' is not valid for storage

### ISSUE THREAD

**Issue by kfern** [feathersjs-ecosystem/feathers-mongodb#101]
Hi! 

When I try to use $inc in the data field of patch method I get "The dollar ($) prefixed field '$inc' in '$inc' is not valid for storage.". Could I incrementing fields using $inc?

### Steps to reproduce

```
  const data = {
    '$inc': {fieldName: 1}
  };
  return app.service('myService').patch(id, data);

```

### System configuration
MongoDB: 3.4
feathers: 2.2.3
feathers-mongodb: 2.9.1

**Comment by kfern** [comment]
Hi!

I think the problem is in [this function](https://github.com/feathersjs/commons/blob/967b70232e4d7e9fdcda4ddd6a13497ad98900e6/lib/filter-query.js#L41
)
```
// Converts Feathers special query parameters and pagination settings
// and returns them separately a `filters` and the rest of the query
// as `query`
module.exports = function (query, paginate) {
  let filters = {
    $sort: convertSort(query.$sort),
    $limit: getLimit(parse(query.$limit), paginate),
    $skip: parse(query.$skip),
    $select: query.$select,
    $populate: query.$populate
  };

  return { filters, query: _.omit(query, ...PROPERTIES) };
};
```

it converts to filters ONLY Feathers officially supported query parameters.

Perhaps the simplest solution would be to move to filter and remove from query all items having the $ sign, Another possibility would be to add a new special parameter to indicate which items in the query are filters.

Would a solution be possible? Increasing a value at the database level guarantees the quality of the information

**Comment by daffl** [comment]
This filters the query, not the data. With MongoDB you should be able to use `$inc` with `update`:

```js
  const data = {
    '$inc': {fieldName: 1}
  };
  return app.service('myService').update(id, data);
```

**Comment by kfern** [comment]
Hello @daffl 

I have executed the following test. 'test-service' is a standard MongoDB service

```
/* eslint-disable no-console */
const logger = require('winston');
const app = require('./app');
const port = app.get('port') + 101;
const server = app.listen(port);

process.on('unhandledRejection', (reason, p) =>
  logger.error('Unhandled Rejection at: Promise ', p, reason)
);


server.on('listening', function (){
  logger.info('Running...');
  setTimeout(() => { 
    run(); 
  }, 1000);//One second delay
});

const run = () => {
  //Delete all collections en test-service
  return app.service('test-service').remove(null, {}).then(() => {
    //Create a new record
    const data = {
      name: 'test_1',
      counter: 1
    }
    return app.service('test-service').create(data).then(result => {
      logger.info('Record created');
      //Update counter field
      const incrementCounter = {
        $inc: {
          counter: 1
        }
      }
      return app.service('test-service').patch(result._id, incrementCounter).then(result => {
        //Result
        logger.info(result);
      })
    })

  }).catch(error => logger.error(error));
  logger.info('Closing...');
  server.close();        
};

```
Should it work or am I doing something wrong? 

```
$ node src/testServiceMongo.js
info: Running...
info: after: test-service - Method: remove
info: after: test-service - Method: create
info: Record created
info: error: test-service - Method: patch: The dollar ($) prefixed field '$inc' in '$inc' is not valid for st
orage.
error:  GeneralError: The dollar ($) prefixed field '$inc' in '$inc' is not valid for storage.

```

**Comment by daffl** [comment]
Closed via #115, thank you @VincentExpoTech
